File: tests/Generator/resource/csharp/complex/ArrayPropertyType.cs
using System.Text.Json.Serialization;

namespace TypeAPI.Model;

/// <summary>
/// Represents an array which contains a dynamic list of values of the same type
/// </summary>
public class ArrayPropertyType : CollectionPropertyType
{
    [JsonPropertyName("type")]
    public new string? Type { get; set; } = "array";

}

File: tests/Generator/resource/python/complex/array_property_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .collection_property_type import CollectionPropertyType


# Represents an array which contains a dynamic list of values of the same type
class ArrayPropertyType(CollectionPropertyType):
    type: Literal["array"] = Field(alias="type")
    pass


